Automattic/jetpack
# [WooCommerce Analytics] Add Common props, new events and bug fixing

## Proposed changes:

* Add a `session_id` as common prop
* Add `session_started` event
* Fix a bug: Fix `add_to_cart` event not being triggered in cart blocks
* Fix a bug: Fix `remove_from_cart` event not being triggered in classic cart page
* Fix a bug: Avoid tracking running in login page and ajax requests.
* Fix a bug: Avoid triggering `checkout_view` when visiting cart page.
* Fix a bug: Fix triggering cart_view doesn't work in  Blocks Cart
* Add common props as order_total, total_shipping, total_discount, total_tax, is_guest, timezone, store_currency and wp_version


### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:

* Prerequisites

1. Download and load Tracks Vigilante Chrome extension https://github.com/Automattic/tracks-chrome-extension
2. Install Woo and Jetpack with this PR
3. Connect Jetpack and WP.com
4. Create a page with Carts Block
5. Create a page with Cart classic
6. Create a page with Checkout Block
7. Create a page with CHeckout classic

* Testing session_started
1. Open an incognito page and navigate to shop
2. Verify `session_started` is tracked in tracks vigilante
3. Verify session_id is added as prop in this and future events

* Testing add_to_cart / remove from cart
5. Add some products to cart (no tracking is added in vigilante at this point)
8. Navigate to a product or refresh the page
9. See previous product_view + add_to_cart event  for previous additions
10. Go to the cart block page
11. See cart_view triggered
12. Remove an item and refresh
13. See remove_from_cart triggered
14. Go to shop again and add more products
15. Open mini cart and remove a product
16. Refresh. See remove_from_cart triggered
17. Go to Cart classic page
18. Remove a product 
19. See remove_from_cart triggered
20. Remove a product by reducing qty as 0 and clicking update
21. See remove_from_cart triggered

* Testing checkout view
23. Go to checkout classic
24. See product_checkout and checkout_view tracked with the right common props
25. Go to checkout blocks
26. See product_checkout and checkout_view tracked with the right common props
27. Complete the order
28. See order_confirmation_view and product_purchase triggered

* Close the incognito window and open a new one

29. Enter the login page
30. Nothing is tracked
31. Enter to Shop in the browser 
32. A new session_started is loaded
